autogen/tests/test.bats

```diff
@@ -0,0 +1,11 @@
+TEST_PKG_VERSION="$(echo "${TEST_PKG_IDENT}" | cut -d/ -f3)"
+
+@test "Autogen version command returns expected version" {
+  result="$(hab pkg exec "${TEST_PKG_IDENT}" autogen --version)"
+  [[ "$result" =~ $TEST_PKG_VERSION  ]]
+}
+
+@test "Autogen usage command returns usage information" {
```

Comment:
Since the first test gives us some assurance that the binary is valid; I don't think this 2nd test is needed.
We have a few core plans where the version or usage output is bugged; but the software is still the version that people need. That might also happen here, if `--usage` breaks people will still need us to provide that update as a core plan.

If you're keen to have a 2nd test, you could maybe try a simple Hello World example to show autogen working in its simplest case. At a glance, autogen doesn't seem to accept input at CLI and you'd need to add some test fixture files for it to read in.
Just having the 1st test is acceptable imo.